tests/unit/test_coresubset.py

```diff
@@ -421,6 +430,343 @@ def mock_kernel_vectorised(_, __):
+        # iterate in blocks of size 0 - so we expect a value error
+        with self.assertRaises(ValueError) as error_raised:
+            herding_object.fit(
+                original_data=self.generic_data,
+                strategy=coreax.reduction.SizeReduce(self.coreset_size),
+            )
+
+        self.assertEqual(
+            error_raised.exception.args[0],
+            "max_size must be a positive integer",
+        )
+
+    def test_kernel_herding_negative_block_size(self):
+        """
+        Test the class KernelHerding when given a negative block_size.
+        """
+        # Define a kernel herding object with the negative valued block_size
+        herding_object = coreax.coresubset.KernelHerding(
+            random_key=self.random_key,
+            kernel=coreax.kernel.SquaredExponentialKernel(),
+            block_size=-5,
+        )
+
+        # The fit method should cap max_size at zero, then not allow uses of zero block
+        # size - since we can't iterate in blocks of size 0 - so we expect a value error
+        with self.assertRaises(ValueError) as error_raised:
+            herding_object.fit(
+                original_data=self.generic_data,
+                strategy=coreax.reduction.SizeReduce(self.coreset_size),
+            )
+
+        self.assertEqual(
+            error_raised.exception.args[0],
+            "max_size must be a positive integer",
+        )
+
+    def test_kernel_herding_float_block_size(self):
+        """
+        Test the class KernelHerding when given a float block_size.
+        """
+        # Define a kernel herding object with the negative valued block_size
+        herding_object = coreax.coresubset.KernelHerding(
+            random_key=self.random_key,
+            kernel=coreax.kernel.SquaredExponentialKernel(),
+            block_size=5.0,
+        )
+
+        # The fit method should reject trying to loop from 0 to a float value of
+        # block_size - raising a type error
+        with self.assertRaises(TypeError) as error_raised:
+            herding_object.fit(
+                original_data=self.generic_data,
+                strategy=coreax.reduction.SizeReduce(self.coreset_size),
+            )
+
+        self.assertEqual(
+            error_raised.exception.args[0],
+            "'float' object cannot be interpreted as an integer",
+        )
+
+    def test_kernel_herding_string_block_size(self):
+        """
+        Test the class KernelHerding when given a string block_size.
+        """
+        # Define a kernel herding object with the negative valued block_size
+        herding_object = coreax.coresubset.KernelHerding(
+            random_key=self.random_key,
+            kernel=coreax.kernel.SquaredExponentialKernel(),
+            block_size="ABC",
+        )
+
+        # The fit method should cap max_size at zero, then not allow uses of zero block
+        # size - since we can't iterate in blocks of size 0 - so we expect a value error
```

Comment:
I suspect the comment here was copied through and then not changed to reflect this test. This is a type error rather than a value error.

Reply:
Thanks for catching - updated